# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../rezound/fox-1.6.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Out of the proaudio SVN repository.

  - Rene Rebe <rene@exactcode.de>

--- rezound/src/frontend_fox/Makefile.am.fox16~	2005-04-30 23:58:16.000000000 +0200
+++ rezound/src/frontend_fox/Makefile.am	2006-03-27 23:40:35.000000000 +0200
@@ -115,14 +115,14 @@
 	$(top_builddir)/src/misc/CNestedDataFile/libmiscCN.la \
 	$(top_builddir)/src/PoolFile/libPoolFile.la
 
-rezound_LDADD=$(_rezound_LDADD) @FOX_1_5_LIBS@  @FOX_1_4_LIBS@ @FOX_1_3_LIBS@ @FOX_1_2_LIBS@ @FOX_1_1_LIBS@ @FOX_LIBS@
+rezound_LDADD=$(_rezound_LDADD) @FOX_1_6_LIBS@ @FOX_1_5_LIBS@  @FOX_1_4_LIBS@ @FOX_1_3_LIBS@ @FOX_1_2_LIBS@ @FOX_1_1_LIBS@ @FOX_LIBS@
 
 ## if I don't do this, automake leaves out @abc@ vars when creating the rezound_DEPENDENCES value
 ## so changes to the LADSPA lib (that may or may not be built) don't cause a re-link
 rezound_DEPENDENCIES=$(_rezound_LDADD)
 
 
-AM_CXXFLAGS=@FOX_1_5_CXXFLAGS@ @FOX_1_4_CXXFLAGS@ @FOX_1_3_CXXFLAGS@ @FOX_1_2_CXXFLAGS@ @FOX_1_1_CXXFLAGS@ @FOX_CXXFLAGS@
+AM_CXXFLAGS=-I/usr/include/fox-1.6 @FOX_1_6_CXXFLAGS@ @FOX_1_5_CXXFLAGS@ @FOX_1_4_CXXFLAGS@ @FOX_1_3_CXXFLAGS@ @FOX_1_2_CXXFLAGS@ @FOX_1_1_CXXFLAGS@ @FOX_CXXFLAGS@
 
 noinst_HEADERS=\
 	fox_compat.h \
--- rezound/configure.ac.fox16~	2005-09-05 17:34:06.000000000 +0200
+++ rezound/configure.ac	2006-03-27 23:28:45.000000000 +0200
@@ -615,19 +615,22 @@
 
 # This is a check for a class in a C++ library. To pinpoint the specific version we need we could change what class is linked.
 # it sets $1_CXXFLAGS and $1_LIBS to either something useful or nothing
-ajv_CXX_CHECK_LIB(FOX-1.5,FXApp,fox-1.5/fx.h,[http://fox-toolkit.org],[],false)
-if test -z "$FOX_1_5_LIBS"; then
-	ajv_CXX_CHECK_LIB(FOX-1.4,FXApp,fox-1.4/fx.h,[http://fox-toolkit.org],[],false)
-	if test -z "$FOX_1_4_LIBS"; then
-		ajv_CXX_CHECK_LIB(FOX-1.3,FXApp,fox-1.3/fx.h,[http://fox-toolkit.org],[],false)
-		if test -z "$FOX_1_3_LIBS"; then
-			ajv_CXX_CHECK_LIB(FOX-1.2,FXApp,fox-1.2/fx.h,[http://fox-toolkit.org],[],false)
-			if test -z "$FOX_1_2_LIBS"; then
-				# libFOX-1.2 was not found, now try libFOX-1.1
-				ajv_CXX_CHECK_LIB(FOX-1.1,FXApp,fox-1.1/fx.h,[http://fox-toolkit.org],[],false)
-				if test -z "$FOX_1_1_LIBS"; then
-					# libFOX-1.1 was not found, now try libFOX
-					ajv_CXX_CHECK_LIB(FOX,FXApp,fox/fx.h,[http://fox-toolkit.org],[],true)
+ajv_CXX_CHECK_LIB(FOX-1.6,FXApp,fox-1.6/fx.h,[http://fox-toolkit.org],[],false)
+if test -z "$FOX_1_6_LIBS"; then
+	ajv_CXX_CHECK_LIB(FOX-1.5,FXApp,fox-1.5/fx.h,[http://fox-toolkit.org],[],false)
+	if test -z "$FOX_1_5_LIBS"; then
+		ajv_CXX_CHECK_LIB(FOX-1.4,FXApp,fox-1.4/fx.h,[http://fox-toolkit.org],[],false)
+		if test -z "$FOX_1_4_LIBS"; then
+			ajv_CXX_CHECK_LIB(FOX-1.3,FXApp,fox-1.3/fx.h,[http://fox-toolkit.org],[],false)
+			if test -z "$FOX_1_3_LIBS"; then
+				ajv_CXX_CHECK_LIB(FOX-1.2,FXApp,fox-1.2/fx.h,[http://fox-toolkit.org],[],false)
+				if test -z "$FOX_1_2_LIBS"; then
+					# libFOX-1.2 was not found, now try libFOX-1.1
+					ajv_CXX_CHECK_LIB(FOX-1.1,FXApp,fox-1.1/fx.h,[http://fox-toolkit.org],[],false)
+					if test -z "$FOX_1_1_LIBS"; then
+						# libFOX-1.1 was not found, now try libFOX
+						ajv_CXX_CHECK_LIB(FOX,FXApp,fox/fx.h,[http://fox-toolkit.org],[],true)
+					fi
 				fi
 			fi
 		fi
@@ -644,6 +647,8 @@
 AC_SUBST(FOX_1_4_LIBS)
 AC_SUBST(FOX_1_5_CXXFLAGS)
 AC_SUBST(FOX_1_5_LIBS)
+AC_SUBST(FOX_1_6_CXXFLAGS)
+AC_SUBST(FOX_1_6_LIBS)
 AC_SUBST(FOX_CXXFLAGS)
 AC_SUBST(FOX_LIBS)
 
--- rezound/src/frontend_fox/fox_compat.h.ark	2006-03-28 00:02:23.000000000 +0200
+++ rezound/src/frontend_fox/fox_compat.h	2006-03-28 00:02:39.000000000 +0200
@@ -21,6 +21,9 @@
 #elif defined(HAVE_LIBFOX_1_5)
 	#include <fox-1.5/fx.h>
 	#include <fox-1.5/fxkeys.h>
+#elif defined(HAVE_LIBFOX_1_6)
+	#include <fox-1.6/fx.h>
+	#include <fox-1.6/fxkeys.h>
 #else
 	#error no HAVE_LIBFOX defined
 #endif
--- rezound/src/frontend_fox/CMainWindow.cpp.ark	2006-03-28 00:15:06.000000000 +0200
+++ rezound/src/frontend_fox/CMainWindow.cpp	2006-03-28 00:15:59.000000000 +0200
@@ -1303,7 +1303,11 @@
 			if(value=="")
 				continue; // no key bound to this action
 
+#ifdef HAVE_LIBFOX_1_6
+			FXHotKey acckey=parseAccel(value.c_str());
+#else
 			FXHotKey acckey=fxparseAccel(value.c_str());
+#endif
 			mc->setAccelText(value.c_str());
 
 			/* these shuttle controls have to be handle specially because they require key up events to return the shuttle to the middle */
@@ -1344,7 +1348,11 @@
 	vector<FXHotKey> removeKeyBindingsList;
 	const vector<string> actionsWithKeyBindings=gKeyBindingsStore->getChildKeys("");
 	for(size_t t=0;t<actionsWithKeyBindings.size();t++)
+#ifdef HAVE_LIBFOX_1_6
+		removeKeyBindingsList.push_back(parseAccel(gKeyBindingsStore->getValue<string>(actionsWithKeyBindings[t]).c_str()));
+#else
 		removeKeyBindingsList.push_back(fxparseAccel(gKeyBindingsStore->getValue<string>(actionsWithKeyBindings[t]).c_str()));
+#endif
 
 	if(gKeyBindingsDialog->showIt(gKeyBindingRegistry))
 	{
--- rezound/src/frontend_fox/FXRezWaveView.cpp.ark	2006-03-28 00:22:20.000000000 +0200
+++ rezound/src/frontend_fox/FXRezWaveView.cpp	2006-03-28 00:22:49.000000000 +0200
@@ -48,7 +48,11 @@
 
 	virtual void create();
 
+#ifdef HAVE_LIBFOX_1_6
+	virtual bool canFocus() const;
+#else
 	virtual FXbool canFocus() const;
+#endif
 
 
 	size_t getClickedCue(FXint x,FXint y);
@@ -369,7 +373,11 @@
 	font->create();
 }
 
+#ifdef HAVE_LIBFOX_1_6
+bool FXWaveRuler::canFocus() const
+#else
 FXbool FXWaveRuler::canFocus() const
+#endif
 {
 	return 1;
 }
--- rezound/src/frontend_fox/CKeyBindingsDialog.cpp.ark	2006-03-28 00:23:09.000000000 +0200
+++ rezound/src/frontend_fox/CKeyBindingsDialog.cpp	2006-03-28 00:23:22.000000000 +0200
@@ -208,7 +208,11 @@
 			}
 			else
 			{
+#ifdef HAVE_LIBFOX_1_6
+				string tmp_key=unparseAccel(keycode).text();
+#else
 				string tmp_key=fxunparseAccel(keycode).text();
+#endif
 				if(tmp_key=="")
 					Error(_("Unhandled Key Combination"));
 				else
